pkg/apis/sources/v1/sinkbinding_lifecycle.go

```diff
@@ -124,8 +125,19 @@ func (sb *SinkBinding) Do(ctx context.Context, ps *duckv1.WithPod) {
 	for i := range spec.InitContainers {
 		spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
 			Name:  "K_SINK",
-			Value: uri.String(),
+			Value: addr.URL.String(),
 		})
+		if addr.CACerts != nil {
+			spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
+				Name:  "K_CA_CERTS",
+				Value: *addr.CACerts,
+			})
+		} else {
+			spec.InitContainers[i].Env = append(spec.InitContainers[i].Env, corev1.EnvVar{
+				Name:  "K_CA_CERTS",
+				Value: "",
+			})
+		}
```

Comment:
I'd not add it when it's not needed / empty
```suggestion
		}
```

Reply:
The benefit is that if someone upgrades to a new version (even with the feature flag disabled), we don't modify any deployment to contain an empty env variable